core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -228,12 +234,13 @@ public void seek(long pos) throws IOException {
       return;
     }
     Preconditions.checkArgument(pos >= 0, PreconditionMessage.ERR_SEEK_NEGATIVE, pos);
-    Preconditions.checkArgument(validPosition(pos),
-        PreconditionMessage.ERR_SEEK_PAST_END_OF_FILE, pos);
-
-    seekBlockInStream(pos);
-    checkAndAdvanceBlockInStream();
-    mCurrentBlockInStream.seek(mPos % mBlockSize);
+    Preconditions.checkArgument(
+        pos < mFileLength, PreconditionMessage.ERR_SEEK_PAST_END_OF_FILE, pos);
```

Comment:
I don't understand. The existing code uses `validPosition(pos)`, but this PR changed the condition. Why is that?